Bugfix/ounp

mickaelaccensi opened the pull request:
**description :**
bugfix for ounp string40
linked to issue #262 

**model :**
ww3_ounp :  correct missing statements for 40 characters names

**matrix :**
done with intel-comp/18, impi/2018.1.163, netCDF-Fortran 4.5.2
all : changes in output netcdf from string16 to string40
mww3_test_03 : not b4b known
[matrix_ounp.zip](https://github.com/NOAA-EMC/WW3/files/5407546/matrix_ounp.zip)

mickaelaccensi committed:
add feature for fixed filename with ndates=0 for ww3_ounf

mickaelaccensi committed:
Merge branch 'feature/fixedname' into 'develop'

fixed file name for ww3_ounf output

See merge request wave/WW3!1

mickaelaccensi committed:
Merge branch 'bugfix/netcdfounf' into 'develop'

bugfix for netcdf file id allocation

See merge request wave/WW3!2

mickaelaccensi committed:
bugfix for netcdf file id allocation

mickaelaccensi committed:
remove extra line due to copy/paste error

mickaelaccensi committed:
Merge branch 'bugfix/lineshel' into 'develop'

extra line in ww3_shel

See merge request wave/WW3!4

mickaelaccensi committed:
Merge branch 'feature/fixedname' into 'develop'

Feature/fixedname

See merge request wave/WW3!5

mickaelaccensi committed:
change vers from 7.00 to 7.01
add comment

mickaelaccensi committed:
Merge remote-tracking branch 'noaa-emc/develop' into develop

mickaelaccensi committed:
add WCOR to switch list

mickaelaccensi committed:
Merge branch 'bugfix/wcor' into 'develop'

add WCOR to switch list

See merge request wave/WW3!6

mickaelaccensi committed:
add check on array index

mickaelaccensi committed:
Merge branch 'bugfix/ugbound' into 'develop'

add check on array index

See merge request wave/WW3!7

mickaelaccensi committed:
revert comment on wcc and wcf in manual

Commit:
Update of IS2 default parameters and associated change to regtest.

Commit:
TEST
Merge branch 'develop' of https://gitlab.ifremer.fr/wave/WW3 into develop

mickaelaccensi committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/WW3 into develop

Commit:
Merge branch 'develop' of https://gitlab.ifremer.fr/wave/WW3 into develop

mickaelaccensi committed:
correct SSDSC array to re-enable T500

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

Commit:
First cleanup before introducing Romero's dissipation

Commit:
Merged STX from Romero into ST4 ... running but not same result yet

Commit:
Fixed bug in Romero option: now working OK.

mickaelaccensi committed:
Merge branch 'feature/ST4tail' of https://gitlab.ifremer.fr/wave/WW3 into feature/ST4tail

Conflicts:
	model/ftn/w3gdatmd.ftn
	model/ftn/w3src4md.ftn
	model/ftn/ww3_grid.ftn

mickaelaccensi committed:
Merge branch 'develop' into feature/ST4tail

mickaelaccensi committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/WW3 into develop

mickaelaccensi committed:
Merge branch 'develop' into feature/ST4tail

Commit:
Minor update

Commit:
Merge branch 'feature/ST4tail' of https://gitlab.ifremer.fr/wave/WW3 into feature/ST4tail

Commit:
Updated the method for computing modulation

Commit:
Bug correction: MSSSUM was not computed in some cases

Commit:
Final but correction for MSSSUM .

Commit:
Corrected one more bug in the MSSSUM calculation ...

Commit:
Added namelist parameters that were missing in SDS4

Commit:
Corrected for bug in additional turbulence dissipation

Commit:
Since I was at it, I added also the effect of water viscosity on the
dissipation

mickaelaccensi committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/WW3 into develop

mickaelaccensi committed:
This reverts commits which are in feature/ST4tail:
82837c726a8b78e5ce630c6450d713b8892b0929
337fe6de91f62e294e36e98d19a8292c53d25848
9816fdb924972e892fde443c438db86eb071ba3b
983f5f6af3f2d72a3f6425c8dac2af3b014fde80
ac5c363e8b7683f8ebda51b659dece0cfded5b7d
21c7316a83e5e5bdfda385eb7a939e2fd9b23268
9b9e2e8713810cc4ec75e3ba8d39a70965a19347
6c42cef9e4b860d3a8cfe6d4781f6b8d40541662
0e0ba5e6915c847716e8860359016693876106ae
85836924313b56aed6357a518b345cced123f80d
8c561707f1a459e7907d7dd89ead3dca53a96fa0

mickaelaccensi committed:
Revert "Update of IS2 default parameters and associated change to regtest."

This reverts commit f2a653b69b1f27fb498434a9ef79e1a0c76edf8a.

mickaelaccensi committed:
correct conflict in previous merge

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
manually merge noaa/develop into develop
to integrate minor changes lost by past mistakes
with git commands

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
Merge remote-tracking branch 'noaa/develop' into develop

mickaelaccensi committed:
correct missing statements for 40 characters names

ukmo-ccbunney reviewed (approved):
Changes look good to me Mickael.

With referecnce to the CHECK_ERR changes, I added the following to ww3_ounf in a branch that I am working on:

    #define CHECK_ERR(I) CHECK_ERROR(I, __LINE__)

and renamed the subroutine to CHECK_ERROR. It saves you from having to provide specific value for ICODE in your modified subroutine, but obviously gives the line number in the pre-processed code (which is fine for  debugging during development). Just thought I would mention it in case you were interested.

mickaelaccensi commented:
> Changes look good to me Mickael.
> 
> With referecnce to the CHECK_ERR changes, I added the following to ww3_ounf in a branch that I am working on:
> 
> ```
> #define CHECK_ERR(I) CHECK_ERROR(I, __LINE__)
> ```
> 
> and renamed the subroutine to CHECK_ERROR. It saves you from having to provide specific value for ICODE in your modified subroutine, but obviously gives the line number in the pre-processed code (which is fine for debugging during development). Just thought I would mention it in case you were interested.

ok, I think we will need to upgrade the CHECK_ERR function to yours in all the WW3 codes

aliabdolali commented:
Thanks @mickaelaccensi 
I also checked it on NOAA HPC with intel, impi, netCDF
[matrix_bugfix_ounp.zip](https://github.com/NOAA-EMC/WW3/files/5429882/matrix_bugfix_ounp.zip)
